src/components/mols/CreditCardPayment/index.tsx

```diff
@@ -0,0 +1,250 @@
+"use client";
+import React, { useState } from "react";
+import { Formik, Form, Field } from "formik";
+import { Input, ButtonPrimary, FormItem, Select } from "@/components";
+import { ICreditCardPayment } from "@/types/payment";
+import * as Yup from "yup";
+import valid from "card-validator";
+import Image from "next/image";
+import CardFlag from "@/images/easytolive/payment/card-flag.svg";
+
+interface ICreditCardPaymentProps {
+  currentStepPayment: React.Dispatch<React.SetStateAction<number>>;
+}
+
+const CreditCardPayment: React.FC<ICreditCardPaymentProps> = ({
+  currentStepPayment,
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [formattedCardNumber, setFormattedCardNumber] = useState("");
+
+  const Months = [
+    " - JAN",
+    " - FEV",
+    " - MAR",
+    " - ABR",
+    " - MAI",
+    " - JUN",
+    " - JUL",
+    " - AGO",
+    " - SET",
+    " - OUT",
+    " - NOV",
+    " - DEZ",
+  ];
+
+  const CreditCardvalidationSchema = Yup.object().shape({
+    cardNumber: Yup.string()
+      .test(
+        "test-number",
+        "Número de cartão inválido",
+        (value) => valid.number(value).isValid,
+      )
+      .required("Digite o número do cartão"),
+    cvv: Yup.string().test(
+      "test-cvv",
+      "CVV inválido",
+      (value) => valid.cvv(value).isValid,
+    ),
+    nameOnCard: Yup.string()
+      .required("Digite o nome do titular")
+      .test(
+        "test-name",
+        "Nome inválido",
+        (value) => valid.cardholderName(value).isValid,
+      ),
```

Comment:
Como regra geral vamo seguir um padrão pra todas as validações que é: `"[Nome do campo] inválido"`. Tanto para vazio quanto para inválido. 


```suggestion
    cardNumber: Yup.string()
      .test(
        "test-number",
        "Número de cartão inválido",
        (value) => valid.number(value).isValid,
      )
      .required("Número de cartão inválido"),
    cvv: Yup.string().test(
      "test-cvv",
      "CVV inválido",
      (value) => valid.cvv(value).isValid,
    ),
    nameOnCard: Yup.string()
      .required("Nome do titular inválido")
      .test(
        "test-name",
        "Nome do titular inválido",
        (value) => valid.cardholderName(value).isValid,
      ),
```